src/main/java/com/powsybl/openloadflow/network/action/LfActionUtils.java

```
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2025, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.network.action;
+
+import com.powsybl.action.*;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.openloadflow.graph.GraphConnectivity;
+import com.powsybl.openloadflow.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+import static com.powsybl.openloadflow.network.action.AbstractLfBranchAction.updateBusesAndBranchStatus;
+
+/**
+ * @author Bertrand Rix {@literal <bertrand.rix at artelys.com>}
+ * @author Anne Tilloy {@literal <anne.tilloy at rte-france.com>}
+ * @author Jean-Luc Bouchot {@literal <jlbouchot at gmail.com>}
+ */
+public final class LfActionUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LfActionUtils.class);
+
+    private LfActionUtils() {
+    }
+
+    public static LfAction createLfAction(Action action, Network network, boolean breakers, LfNetwork lfNetwork) {
```

Why don(t you return an optional ? The previous code could help detect NOOP actions for the network by returning an empty value if the element did not exist.
You now loose the ability to detect actions that do nothing on a given LfNetwork.

---

The old code was returning an optional but actually did not do anything with the "empty" information and just silently ignored the invalid action.
Same checks are done in the apply method with a boolean return and a warning is logged (to be discussed maybe).
Ultimately I think the operator strategy result API should hold the info of what action was applied successfully, this is also why I designed it like this, but this a change of API to be seen in the futur..

---

In the current code, even if the list of lfAction converted from iidm actions is empty there is always a load flow done and an operator strategy result is produced (see AbstractSecurityAnalysis.runSimulations usage of lfActionById).
Whether that is a good thing is probably a good question but there is no functionnal change here as far as I can see.

Furthermore we still have the checkActions method that do some additional checks and raise an exception if there is some inconsistency. This one is still there.

Ok for putting the missing action in the report, good idea!

---

Oh... You a right ! So let-s consider the fact that the LF is run even when there is no action something not ideal, but independant from this work....

I'll close that thread when the report is there.